### ISSUE TITLE
Elementor NavMenu Drop-Down Alignment

### ISSUE THREAD

**Issue by dramirvf** [pojome/elementor#4991]
Hello,
There is an annoying default alignment is set for Elementor Dropdown which is left alignment.
Additionally, there is no option to set the width of dropdown menu. It is better to set the width according to the width of parent menu item. With the current setting, if the parent menu item is has more letters (which results in a bigger length), the dropdown will align to its left, with a fixed width which is very ugly.

Customer service of Elementor Pro has referred me to post here so Elementor Developers could look into it.

Here is the snapshot:
![dropdown 1](https://user-images.githubusercontent.com/40552131/42418033-78861fe2-8265-11e8-80b1-71d9132d15b8.jpg)

**There could be 2 solutions:**
1. Set the default width of dropdown menu as the vale to the current width of parent menu item.
2. Set the alignment of dropdown menu as "Center" not "left".

Is there a way I manually change the setting in the relevant js file at the moment?

**Thanks,
Amir**

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future.

I merged this issue with  #3275, which is on the same subject (nav menu customizations) so we could deal with it more efficiently. 
Please feel free to continue the conversation there.

Thanks for your feedback!

**Comment by davidofug** [comment]
Was this issue worked on?
Thank you

**Comment by desparkflux** [comment]
Please how do I make menu nav bar... How do I integrate it on my web?

**Comment by djnolder** [comment]
A temporary solution for anyone finding this...

your nav menu > Advanced > Advanced > CSS Classes = "full_center_dropdown"
your nav menu > Advanced > Custom CSS = 
.full_center_dropdown nav ul {
    min-width: 100% !important;
}
.full_center_dropdown nav ul a {
    justify-content: center;
}

This will give your submenus a minwidth of 100%, and will center the menu items. If you want to right-justify the menu items change "justify-content: center;" to "justify-content: flex-end;"

Hopefully, we will one day git this update.

**Comment by gautvd** [comment]
> A temporary solution for anyone finding this...
> 
> your nav menu > Advanced > Advanced > CSS Classes = "full_center_dropdown"
> your nav menu > Advanced > Custom CSS =
> .full_center_dropdown nav ul {
> min-width: 100% !important;
> }
> .full_center_dropdown nav ul a {
> justify-content: center;
> }
> 
> This will give your submenus a minwidth of 100%, and will center the menu items. If you want to right-justify the menu items change "justify-content: center;" to "justify-content: flex-end;"
> 
> Hopefully, we will one day git this update.

This works perfectly ! Thanks !

You, Sir, go up !

**Comment by wafiqsyed** [comment]
@djnolder That worked perfectly and improved my site design big time! Thank you

**Comment by maurocarreraok** [comment]
Same issue here, but the fix is not working :(

Can you help me with this?

![image](https://user-images.githubusercontent.com/68023943/93362339-3f1f2580-f81c-11ea-963a-9df25af69b32.png)

As you can see, i need the submenu FOR HIM FOR HIS centered below the SHOP menu... 

Thanks!!!

**Comment by cameronmiller1988** [comment]
> Same issue here, but the fix is not working :(
> 
> Can you help me with this?
> 
> ![image](https://user-images.githubusercontent.com/68023943/93362339-3f1f2580-f81c-11ea-963a-9df25af69b32.png)
> 
> As you can see, i need the submenu FOR HIM FOR HIS centered below the SHOP menu...
> 
> Here is the URL: https://srfreeck.com/en
> 
> Thanks in advance!!! @djnodler

You may need to reduce the padding or play with the min width to make it wider than the parent nav.

**Comment by ikoba218** [comment]
> Same issue here, but the fix is not working :(
> 
> Can you help me with this?
> 
> ![image](https://user-images.githubusercontent.com/68023943/93362339-3f1f2580-f81c-11ea-963a-9df25af69b32.png)
> 
> As you can see, i need the submenu FOR HIM FOR HIS centered below the SHOP menu...
> 
> Here is the URL: https://srfreeck.com/en
> 
> Thanks in advance!!! @djnodler

If there isn't any set margin or padding as @cameronmiller1988 said, then it may be easy to fix with a custom CSS. But I can't see your website, it is not loading.

**Comment by Bovelett** [comment]
@djnolder Thank you for this solution, works like a charm! 🙏

**Comment by samuelgfeller** [comment]
@djnolder solution did not work for me but got me inspired for the following which did work for me (without custom css class but taking elementor's default one instead):
```css
.elementor-nav-menu--dropdown{
    min-width: 100% !important;
}
```

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce the new Menu widget released with Elementor Pro v3.12.0! 🥳 

With the new (nested) Menu widget, you can customize your menus better than ever! 💪 
Drag & Drop widgets to create complex Mega Menus, fine-tune your menu designs, and add relevant information.

This is just the first phase. More customization is coming!

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
